src/metabase/driver/postgres.clj

```diff
@@ -44,9 +44,13 @@
   (hx/cast-unless-type-in "timestamp" #{"timestamp" "timestamptz" "date"} honeysql-form))
 
 (defmethod sql.qp/add-interval-honeysql-form :postgres
-  [_ hsql-form amount unit]
-  (hx/+ (->timestamp hsql-form)
-        (hsql/raw (format "(INTERVAL '%s %s')" amount (name unit)))))
+  [driver hsql-form amount unit]
+  ;; Postgres doesn't support quarter in intervals (#20683)
+  (if (= unit :quarter)
+    (recur driver hsql-form (* 3 amount) :month)
+    (let [hsql-form (->timestamp hsql-form)]
+      (-> (hx/+ hsql-form (hsql/raw (format "(INTERVAL '%s %s')" amount (name unit))))
+          (hx/with-type-info (hx/type-info hsql-form))))))
```

Comment:
The `with-type-info` is also not strictly part of this fix but this tweak lets us avoid more unneeded `cast(... AS timestamp)` casts since a `timestamp` or interchangeable plus a date interval doesn't need to be casted again after adding the interval. Probably not a noticeable difference performance wise but it simplifies the SQL we generate which is still a win.